reframe/frontend/cli.py

```diff
@@ -410,15 +411,27 @@ def main():
 
     # Setup the check loader
     if options.checkpath:
-        load_path = []
+        load_path = OrderedSet()
         for d in options.checkpath.split(':'):
             d = os_ext.expandvars(d)
             if not os.path.exists(d):
                 printer.warning("%s: path `%s' does not exist. Skipping..." %
                                 (argparser.prog, d))
                 continue
 
-            load_path.append(d)
+            d = os.path.abspath(os.path.realpath(d))
+            if d not in load_path:
+                load_path.add(d)
```

Comment:
```python
if d in load_path:
     continue
```

And then add it to the path after you have done the analysis for the recursive option below, which should be done in this loop, on `d`, not in a separate one.

Reply:
The problem with this is that the shorter path may come after the longer one. So if '/a/b/c' preceeds '/a/'b' then I will not have considered it in the search for the parent paths of '/a/b/c'.